pkg/config/loader.go

```diff
@@ -0,0 +1,93 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/ipfs/testground/pkg/logging"
+)
+
+const (
+	EnvTestgroundHomeDir = "TESTGROUND_HOME"
+
+	DefaultListenAddr = "localhost:8042"
+)
+
+func (e *EnvConfig) Load() error {
+	// apply fallbacks.
+	e.Daemon.Listen = DefaultListenAddr
+	e.Client.Endpoint = DefaultListenAddr
+
+	// calculate home directory; use env var, or fall back to $HOME/testground
+	// otherwise.
+	var home string
+	if v, ok := os.LookupEnv(EnvTestgroundHomeDir); ok {
+		// we have an env var.
+		home = v
+	} else {
+		// fallback to $HOME/testground.
+		v, err := os.UserHomeDir()
+		if err != nil {
+			return fmt.Errorf("failed to obtain user home dir: %w", err)
+		}
+		home = filepath.Join(v, "testground")
+	}
+
+	switch fi, err := os.Stat(home); {
+	case os.IsNotExist(err):
+		logging.S().Infof("creating home directory at %s", home)
+		if err := os.MkdirAll(home, 0777); err != nil {
+			return fmt.Errorf("failed to create home directory at %s: %w", home, err)
+		}
+	case err == nil:
+		logging.S().Infof("using home directory: %s", home)
+	case !fi.IsDir():
+		return fmt.Errorf("home path is not a directory %s", home)
+	}
+
+	// ensure home and children directories exist.
+	e.dirs = Directories{home}
+	for _, d := range []string{
+		e.dirs.Home(),
+		e.dirs.Outputs(),
+		e.dirs.Plans(),
+		e.dirs.SDKs(),
+		e.dirs.Work(),
+	} {
+		if err := ensureDir(d); err != nil {
+			return fmt.Errorf("failed to check/create directory %s: %w", d, err)
+		}
+	}
+
+	// parse the .env.toml file, if it exists.
+	f := filepath.Join(e.dirs.Home(), ".env.toml")
+	if _, err := os.Stat(f); err == nil {
+		// try to load the optional .env.toml file
+		_, err = toml.DecodeFile(f, e)
+		if err != nil {
+			return fmt.Errorf("found .env.toml at %s, but failed to parse: %w", f, err)
+		}
+		logging.S().Infow(".env.toml loaded from %s", f)
```

Comment:
`Infof`

Also, `testground daemon` now complains with:

```
Apr 16 12:14:45.923955  DPANIC  Ignored key without a value.    {"ignored": "/Users/nonsense/testground/.env.toml"}
Apr 16 12:14:45.923946  INFO    .env.toml loaded from %s
```

I am not sure what this DPANIC is about, because I don't have keys without values in my `.env.toml`.

Reply:
It's coming from zap, due to the wrong log format.

Reply:
Fixed in https://github.com/ipfs/testground/pull/849/commits/6d6296cd161a91c89b1f7f624cc519abbca814de